lib/Cake/Network/CakeSocket.php

```diff
@@ -129,8 +129,12 @@ public function connect() {
 			$this->disconnect();
 		}
 
+		$hasProtocol = strpos($this->config['host'], '://') !== false;
```

Comment:
Why isn't this using `parse_url`?

I.e.:

![screenshot from 2015-12-01 15-03-38](https://cloud.githubusercontent.com/assets/33387/11502721/ce23c834-983c-11e5-9086-9a0708da89e5.png)